### PR TITLE
fix: ensure messages are only shown once

### PR DESCRIPTION
**Pull request by stephen-pope-customer-io**
The idea is to keep track of the shown messages by queueId (which is unique per message) and skip the shown ones

This is based on the same solution applied for gist-web: https://github.com/customerio/gist-web/pull/26

Task: https://github.com/customerio/issues/issues/11967

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-android/develop/docs/dev-notes/GIT-WORKFLOW.md).
- [x] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request.
- [x] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1).
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request and it will automatically be merged.

### CODE REVIEW

File: messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt [hunk 1]
```diff
@@ -145,15 +146,26 @@ object GistSdk : Application.ActivityLifecycleCallbacks {
...
+                    messageShown = gistModalManager.showModalMessage(message, position)
+                } catch (e: Exception) {
+                    Log.e(GIST_TAG, "Failed to show message: ${e.message}", e)
+                }
             }
         }
-        return if (messageShown) message.instanceId else null
+
+        if (messageShown) {
+            shownMessageQueueIds.add(queueId!!)
```
Comment (BernardGatt) [inline messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt:164]: Have you tested launching a message programmatically? Wouldn't this crash the app if queueId is not present?

**Comment by BernardGatt** [comment]
I think it would be simpler if the implementation was handled inside `Queue.kt` instead of `GistSdk.kt`

**Comment by BernardGatt** [comment]
This has been updated and tested, ready for review.

**Review by Shahroz16 — APPROVED** [review]
Looks good, only thing would be us not clearing the queue, don't think it can be an issue but just raising it.

**Comment by BernardGatt** [comment]
Yeah, we should eventually implement a retry mechanism on failure and maybe persist this variable into user preferences.

**Pull request merged**